readthedocs/proxito/middleware.py

```diff
@@ -83,6 +83,12 @@ def map_host_to_project_slug(request):  # pylint: disable=too-many-return-statem
         project_slug = domain.project.slug
         request.cname = True
         log.debug('Proxito CNAME: host=%s', host)
+
+        if domain.https and not request.is_secure():
+            # Redirect HTTP -> HTTPS (302) for this custom domain
+            log.debug('Proxito CNAME HTTPS Redirect: host=%s', host)
+            return redirect('https://{}{}'.format(host, request.get_full_path()))
```

Comment:
If possible, I'd like to see a custom RTD header on here so we know where this request comes from similar to https://github.com/readthedocs/readthedocs.org/blob/d030d67b4ba83974b2d5df16b5bd71254487ebaf/readthedocs/proxito/views/mixins.py#L179

Reply:
I think I'll do `X-RTD-Redirect` and the value will be `https`. I might use the same header with the value `canonicalize` when doing the domain canonicalization.